### PR TITLE
Implemented get user custodianship status

### PR DESCRIPTION
**Pull request by IgorGnusFreja**

### CODE REVIEW

**Review by milicajelic-frejaeid — CHANGES_REQUESTED** [review]

File: FrejaEidClient/src/main/java/com/verisec/frejaeid/client/service/RequestValidationService.java [hunk 1]
```diff
@@ -231,4 +232,20 @@ private void validateAdvancedSignRequestedAttributes(SignatureType signatureType
...
 
+    public void validateGetUserCustodianshipStatusRequest(
+            GetUserCustodianshipStatusRequest getUserCustodianshipStatusRequest) throws FrejaEidClientInternalException {
+        String relyingPartyId = getUserCustodianshipStatusRequest.getRelyingPartyId();
+        if (relyingPartyId != null) {
+            if (StringUtils.isEmpty(relyingPartyId)) {
+                throw new FrejaEidClientInternalException("RelyingPartyId cannot be empty.");
+            }
+        }
+        if(StringUtils.isEmpty(getUserCustodianshipStatusRequest.getUserCountryIdAndCrn()) ||
```
[the remark this answers is not in the document]
Comment (IgorGnusFreja) [reply FrejaEidClient/src/main/java/com/verisec/frejaeid/client/service/RequestValidationService.java:243]: Done

**Review by milicajelic-frejaeid — APPROVED** [review]

File: FrejaEidClient/src/main/java/com/verisec/frejaeid/client/beans/custodianship/get/GetUserCustodianshipStatusRequestBuilder.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
+package com.verisec.frejaeid.client.beans.custodianship.get;
+
+public class GetUserCustodianshipStatusRequestBuilder {
+    public static class SetParamsBuilder {
+
+        /**
+         * Sets country code ID and CRN of user
+         *
+         * @param countryIdAndCrn            will be used to identify the end user. Must start with country
```
Comment (milicajelic-frejaeid) [inline FrejaEidClient/src/main/java/com/verisec/frejaeid/client/beans/custodianship/get/GetUserCustodianshipStatusRequestBuilder.java:9]: Formate code.

File: FrejaEidClient/src/main/java/com/verisec/frejaeid/client/enums/UserCustodianshipStatusResult.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ * @author veiggnu
+ */
+public enum UserCustodianshipStatusResult {
+    /**
+     * Custodianship status when the user has a custodian.
+     */
+    USER_HAS_CUSTODIAN("USER_HAS_CUSTODIAN"),
+    /**
+     * Custodianship status when we do not have a registered custodian
```
Comment (milicajelic-frejaeid) [inline FrejaEidClient/src/main/java/com/verisec/frejaeid/client/enums/UserCustodianshipStatusResult.java:13]: Custodianship status when it's unknown if the user has a custodian.

Comment (IgorGnusFreja) [reply FrejaEidClient/src/main/java/com/verisec/frejaeid/client/enums/UserCustodianshipStatusResult.java:13]: Done

File: FrejaEidClient/src/main/java/com/verisec/frejaeid/client/beans/custodianship/get/GetUserCustodianshipStatusRequestBuilder.java (hunk 2 above)
Comment (IgorGnusFreja) [reply FrejaEidClient/src/main/java/com/verisec/frejaeid/client/beans/custodianship/get/GetUserCustodianshipStatusRequestBuilder.java]: Done

**Review by milicajelic-frejaeid — APPROVED** [review]

**Pull request merged**